symphony-bdk-core/src/main/java/com/symphony/bdk/core/config/model/BdkExtAppConfig.java

```diff
@@ -1,10 +1,10 @@
 package com.symphony.bdk.core.config.model;
 
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+
 import lombok.Getter;
 import lombok.Setter;
 
-import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
-
 @Getter
 @Setter
 public class BdkExtAppConfig {
```

Comment:
`BdkExtAppConfig` looks like `BdkBotConfig` at some points, right?

Reply:
There is a small difference in field name between appId and username. For the others, it seems similar.

Reply:
It could be great to have the common fields and common methods factorized in a base class, esp. `isCertificateAuthenticationConfigured()`, `certificatePath` and `certificatePassword`.
And it would enable you to extract the logic in ApiClientFactory lines 98 and 115